lib/dialects/sqlite3/schema/ddl.js

```diff
@@ -58,50 +58,52 @@ class SQLite3_DDL {
   }
 
   getTableSql() {
-    this.trx.disableProcessing();
-    return this.trx.raw(getTableSql(this.tableName())).then((result) => {
-      this.trx.enableProcessing();
-      return result;
-    });
+    return this.client.transaction(
```

Comment:
Wouldn't this result in starting one more transaction on top of already running one?

Reply:
ah, I see you just pushed existing transaction creation logic slightly deeper.

Reply:
Exactly, I had to do this because enabling/disabling the foreign keys check inside a transaction does not work. But the transaction is still needed to rollback in case the check fails.